pkg/scheduler/objects/preemption.go

```diff
@@ -153,6 +155,7 @@ func (p *Preemptor) initWorkingState() {
 			}
 			allocationsByNode[nodeID] = append(allocations, allocation)
 			queueByAlloc[allocation.GetAllocationKey()] = victims
+			nodeInstanceTypeMap[nodeID] = allocation.GetInstanceType()
```

Comment:
This *should* work in theory, however, it's probably best to check that allocation.GetInstanceType() was in fact populated. If not, ignore it and move on. We can use unknown if the map entry ends up not being populated.

Reply:
We should only set the node value if the allocation was populated. If the instance type is empty, we might clobber a valid instance type from another allocation.